<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="joy_dev" default="/dev/input/js0" />

  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="dev" value="$(arg joy_dev)" />
    <param name="deadzone" value="0.1" />
    <param name="autorepeat_rate" value="20" />
  </node>

  <include file="$(find ar_sys)/launch/single_board_blackfly.launch" />

  <node pkg="tud_momana" name="joy_control" type="joy_control" output="screen"/>

  <node pkg="tud_momana" name="moma_odom" type="moma_odom"  output="screen"/>

  <!-- static transforms needed for our relative odometry frames -->
  <!-- map to odom -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="odom_tf" args="0 0 0 0 0 0 1 map odom" />

  <!-- odom to rel_odom -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="rel_odom_tf" args="0 0 0 0 0 0 1 odom rel_odom" />

  <!-- rel_odom to rel_c3po/odom-->
  <node pkg="tf2_ros" type="static_transform_publisher" name="rel_c3po_odom_tf" args="0 0 0 0 0 0 1 rel_odom rel_cam/odom" />

  <!-- rel_odom to rel_r2d2/odom-->
  <node pkg="tf2_ros" type="static_transform_publisher" name="rel_r2d2_odom_tf" args="0 0 0 0 0 0 1 rel_odom rel_marker/odom" />

</launch>
